### PR TITLE
Organization create/modify/delete logic

### PR DESCRIPTION
**Pull request by ryanpbrewster**
@Colin-Lane @leogrim @yingjieMiao 

Cleaned up, passes tests. I'll merge unless y'all have objections.

### CODE REVIEW

File: kifi-backend/modules/shoebox/app/com/keepit/commanders/OrganizationCommander.scala [hunk 1]
```diff
@@ -2,19 +2,117 @@ package com.keepit.commanders
...
+        permissions.exists { _.contains(EDIT_ORGANIZATION) } && !hasInvalidModification(modifications)
+      case OrganizationDeleteRequest(orgId, requesterId) =>
+        requesterId == orgRepo.get(orgId).ownerId
+    }
+  }
+  private def hasInvalidModification(modifications: OrganizationModifications): Boolean = {
+    lazy val badName = modifications.newName.exists(_.isEmpty)
+    lazy val badBasePermissions = modifications.newBasePermissions.exists { bps =>
+      // Are there any members that can't even see the organization?
+      !OrganizationRole.all.forall { role => bps.forRole(role).contains(VIEW_ORGANIZATION) }
```
Comment (Colin-Lane) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/OrganizationCommander.scala]: the double negative made me think twice about this since we are checking for `invalid` modifications. Would it be better to rename the method to `hasValidModifications` and invert the conditions.

Comment (Colin-Lane) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/OrganizationCommander.scala]: We could even name it `isValidModificationRequest`

Comment (ryanpbrewster) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/OrganizationCommander.scala]: I originally had it that way, but I inverted it because the logic was kind of hard to follow.

The condition for being invalid is a little easier to think about, since the no-op is automatically valid. Thus, we only care about changes that both EXIST and are INVALID.

Comment (leogrim) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/OrganizationCommander.scala]: Did you see my comments about modifications? I would expected users to update the organization and "submit" their changes altogether, so I don't know if individual modifications make sense.

Comment (ryanpbrewster) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/OrganizationCommander.scala]: The `modifyOrganization` method takes in a whole list of modifications.

Comment (ryanpbrewster) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/OrganizationCommander.scala]: Rather, it takes in an `OrganizationModifications` object, which encompasses all the fields that can be edited in an `Organization`, and checks each modification one-by-one to ensure they are all valid modifications.
